website/docs.md

```diff
@@ -1986,6 +1986,350 @@ with the same arg value`.
+| storageSpec    | optional  | Options to configure Kubernetes [volumes][kubernetes_volumes] and [volume mounts][kubernetes_volumes]. If absent, no storage is configured.                                                                                                                                              |
+| useHostNetwork | optional  | If true, application listeners use the underlying nodes' network. This behavior is generally discouraged, but it may be useful when running the application in a minikube environment.                                                                                                   |
+
+For more details on specific subfields of each configuration knob, please check
+all the [configuration options][kube_config_file].
+
+**Note**: Configuration knobs such as `resourceSpec`, `scalingSpec`, `storageSpec`
+can be configured both per deployment and per group of colocated components.
+However, if a field has definitions both per deployment and per group, the `KUBE`
+deployer will consider the per group value of the field (except for the `storageSpec`
+where it considers the concatenation of both). For example in the example below,
+the `KUBE` deployer will run two colocation groups, where the pods that run the
+`Reverser` component require at least `256Mi` memory while the pods that run the
+`Main` component require at least `64Mi` memory.
+
+```yaml
+appConfig: weaver.toml
+repo: docker.io/mydockerid
+
+listeners:
+- name: hello
+  public: true
+
+resourceSpec:
+  requests:
+    memory: "64Mi"
+
+groups:
+  - name: reverser-group
+    components:
+      -  github.com/ServiceWeaver/weaver/examples/hello/Reverser
+    resourceSpec:
+      requests:
+        memory: "256Mi"
+```
+
+## Telemetry
+
+The `KUBE` deployer allows you to customize how to export logs, metrics and
+traces. To do that, you need to implement a wrapper deployer on top of the
+`KUBE` deployer using the [KUBE tool][kube_telemetry_api] abstraction.
+
+[Here][kube_telemetry] is an example on how we export metrics to [Prometheus][prometheus] and traces to
+[Jaeger][jaeger].
+
+For example, to export traces to Jaeger, you have to do the following:
+1. Deploy [Jaeger][jaeger] in the Kubernetes cluster as typical Kubernetes services.
+This is what someone will have to do in practice as well.
+```console
+$ kubectl apply -f jaeger.yaml
+```
+2. Write a simple binary that implements the plugin to export traces to Jaeger. The code looks as follows:
+
+```go
+// ./examples/customkube
+...
+
+const jaegerPort = 14268 // Port on which the Jaeger service is receiving traces.
+
+func main() {
+  // Invokes the `KUBE` deployer with the plugin to export traces as instructed
+  // by handleTraceSpans.
+  tool.Run("deploy", tool.Plugins{
```

Comment:
This call to `tool.Run` has to be below the definition of `handleTraceSpans`. I also realized that I used the name `"deploy"` here, but that was a bad name. It should be `"customkube"` in this example.

Reply:
Done